include/solvers/nonlinear_solver.h

```diff
@@ -349,6 +349,16 @@ class NonlinearSolver : public ReferenceCountedObject<NonlinearSolver<T>>,
    */
   bool converged;
 
+  /**
+   * Whether we should reuse the linear preconditioner
+   */
+  bool reuse_preconditioner;
```

Comment:
Can we use getters and setters (defaulting to `libmesh_not_implemented()` outside of PETSc) rather than public member access for these new variables?  It would be nice if, when someone tries to use this feature with Nox or with future subclasses that don't support it right away, they get an error message rather than getting silently ignored.

Reply:
Yep, good idea.

Reply:
Yep, good idea.